### ISSUE TITLE
probe2WayMC() does not work when latent means are fixed to zero

### ISSUE THREAD

**Issue by yrosseel** [simsem/semTools#108]
The example of probe2WayMC() frees the latent means. Perhaps this is by design, but this is not clearly indicated in the man page. Failing to free the latent means currently (0.5-5) leads to the error message:

```
Error in varEst[targetcol, targetcol] : subscript out of bounds
```

which seems to be due to paste() (line 1116) adding extra spaces. Using sep = "" or using paste0() may do the trick.

Below is a reprex:

```
library(semTools)

dat2wayMC <- indProd(dat2way, 1:3, 4:6)

model2 <- '
    f1 =~ x1 + x2 + x3
    f2 =~ x4 + x5 + x6
    f12 =~ x1.x4 + x2.x5 + x3.x6
    f3 =~ x7 + x8 + x9
    f3 ~ f1 + f2 + f12
    f12 ~~ 0*f1 + 0*f2
'

fitMC2way2 <- sem(model2, data = dat2wayMC, meanstructure = TRUE)

probe2WayMC(fitMC2way2, nameX = c("f1", "f2", "f12"), nameY = "f3",
            modVar = "f2", valProbe = c(-1, 0, 1))
```

Yves.

**Comment by emvira** [comment]
Hi Yves, 
I've encountered the same error message. I tried your example with my own data, but the same error message came up. Is there another way around this error? I believe it's been fixed in older versions of semTools.

Below is my example: 

`ind.protective =~ Hope_T3 + Self_Efficacy_T3 + + T3_Self_esteem 

puberty =~ T3_length + T3_hair + T3_skin + T3_breast + T3_period

difficulties =~ T3_SDQ_1 + T3_SDQ_7 + T3_SDQ_12 + T3_SDQ_20_R + T3_SDQ_4 + T3_SDQ_15 + T3_SDQ_3 + T3_SDQ_14 + T3_SDQ_2 + T3_SDQ_6 + T3_SDQ_10 + T3_SDQ_13 + T3_SDQ_19

ind.moderator =~ T3_length.T3_Self_esteem + T3_length.Hope_T3 + T3_length.Self_Efficacy_T3 + T3_hair.T3_Self_esteem + T3_hair.Hope_T3 + T3_hair.Self_Efficacy_T3 + T3_skin.T3_Self_esteem + T3_skin.Hope_T3 + T3_skin.Self_Efficacy_T3 + T3_breast.T3_Self_esteem + T3_breast.Hope_T3 + T3_breast.Self_Efficacy_T3 + T3_period.T3_Self_esteem + T3_period.Hope_T3 + T3_period.Self_Efficacy_T3

ind.moderator ~~ 0*puberty + 0*ind.moderator

#Regressions
difficulties ~ puberty + ind.protective + ind.moderator

mod.ind.fit <- sem(ind.mod,
                   girls.mod,
                   std.lv = T,
                   meanstructure = T,
                   estimator = "MLR")

probe1 <- probe2WayMC(mod.ind.fit, nameX = c("puberty", "ind.protective", "ind.moderator"), nameY = "difficulties", modVar = "ind.protective", valProbe = c(-1, 0, 1))`

**Comment by TDJorgensen** [comment]
Sorry for the delay, finally had time to investigate.

> seems to be due to paste() (line 1116) adding extra spaces. Using sep = "" or using paste0() may do the trick

@yrosseel I suppose you were looking at source code from the CRAN version, because it turns out I already fixed this, only a few days after sending 0.5-5 to CRAN last July (face-palm) 4fb1183

> Could you illustrate where you would add the argument in the code?

@emvira This is something for me to fix in `semTools`, not for the user.  You can install the development version (see syntax on the Wiki tab), but I should be sending a new version to CRAN this week.
